### PR TITLE
feat: generate consistent QueryObject whether GenericAxis is enabled or disabled

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

This PR intends to generate a consistent QueryObject whether GenericAxis FF is enabled or disabled. e.g.: the charts wouldn't change on the Dashboard although Axis FF from `enabled` to `disabled` and vice versa.

1. Make sure that the charts on the Dashboards are consistent whether FF is enabled or disabled, so the FF GenericAxis should be safety enabled by default (will do in separated PR).
2. Add unit test in Echart timeseries charts.
3. Add unit test in Echart MixedTimeseries charts.
4. Move `getXAxis` and `isXAxisSet` into `superset-ui/core`

There are 2 kinds of `queryObject` patterns after this PR:

#### The QueryObjec will be like below when the `x_axis` is not in formData whether `GenericAxis` is enabled or disabled
```
{
        granularity: 'time_column',
        extras: { time_grain_sqla: 'P1Y', having: '', where: '' },
        is_timeseries: true,
        post_processing: [
          {
            operation: 'pivot',
            options: {
              aggregates: { 'count(*)': { operator: 'mean' } },
              columns: ['col1'],
              drop_missing_columns: true,
              index: ['__timestamp'],
            },
          },
          { operation: 'flatten' },
        ],
      }
     ....
     ....
}
```

#### The QueryObjec will be like below when the `x_axis` is in formData whether `GenericAxis` is enabled or disabled.
The `granularity`(axis) and `time_grain` will be converted to `BASE_AXIS`, then remove `time_grain_sqla` and `is_timeseries`

```
{
        extras: { having: '', where: '' },
        columns: [
          {
            columnType: 'BASE_AXIS',
            expressionType: 'SQL',
            label: 'time_column',
            sqlExpression: 'time_column',
            timeGrain: 'P1Y',
          },
          'col1',
        ],
        post_processing: [
          {
            operation: 'pivot',
            options: {
              aggregates: { 'count(*)': { operator: 'mean' } },
              columns: ['col1'],
              drop_missing_columns: true,
              index: ['time_column'],
            },
          },
          { operation: 'flatten' },
        ],
      }
     ....
     ....
```

#### Test Case 1
1. Turn off GenericAxis
2. Make an Echart time-series chart and a Mixed time-series, save to a new Dashboard.
3. Turn on GenericAxis
4. Refresh Dashboard
5. Charts should be as before

#### Test Case 2
1. Turn on GenericAxis
2. Make an Echart time-series chart and a Mixed time-series, save to a new Dashboard.
3. Turn off GenericAxis
4. Refresh Dashboard
5. Charts should be as before

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces a new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/packages/superset-ui-core/src/query/getAxis.ts [hunk 1]
```diff
@@ -16,20 +16,18 @@
...
  */
-import {
-  DTTM_ALIAS,
-  getColumnLabel,
-  isDefined,
-  QueryFormData,
-} from '@superset-ui/core';
+import { DTTM_ALIAS, getColumnLabel, QueryFormData } from '@superset-ui/core';
+
+export const isEnabledAxes = (formData: QueryFormData) => !!formData.x_axis;
```
Comment (villebro) [inline superset-frontend/packages/superset-ui-core/src/query/getAxis.ts:21]: Could we rename this to `isXAxisEnabled` or better yet `isXAxisSet` for clarity?

**Review by villebro — COMMENTED** [review]
Awesomen work, looks great! One minor comment, I'll test this later tonight EET.

File: superset-frontend/packages/superset-ui-core/src/query/getAxis.ts (hunk 1 above)
Comment (zhaoyongjie) [reply superset-frontend/packages/superset-ui-core/src/query/getAxis.ts:21]: Make sense, thanks.

**Review by villebro — APPROVED** [review]
This is really great work @zhaoyongjie, thanks for all these improvements! While testing I looked at the backend logs, and noticed one last deprecation notice:
```
2022-09-20 20:32:39,010:WARNING:superset.common.query_object:The field `timeseries_limit` is deprecated, please use `series_limit` instead.
```
This PR is already pretty big, so no need to fix this here, but once we replace `timeseries_limit` with `series_limit` we should be rid of all deprecation notices for the main ECharts viz types 👍

**Comment by zhaoyongjie** [comment]
> This is really great work @zhaoyongjie, thanks for all these improvements! While testing I looked at the backend logs, and noticed one last deprecation notice:
> 
> ```
> 2022-09-20 20:32:39,010:WARNING:superset.common.query_object:The field `timeseries_limit` is deprecated, please use `series_limit` instead.
> ```
> 
> This PR is already pretty big, so no need to fix this here, but once we replace `timeseries_limit` with `series_limit` we should be rid of all deprecation notices for the main ECharts viz types 👍

Thanks @villebro. I will fix it in a separated PR.

**Pull request merged**